lldb/scripts/framework-header-fix.py

```diff
@@ -0,0 +1,117 @@
+
+
+def modify_rpc_includes(input_directory_path, output_directory_path):
+    for input_filepath in os.listdir(input_directory_path):
+        current_input_file = os.path.join(input_directory_path, input_filepath)
+        output_dest = os.path.join(output_directory_path, input_filepath)
+        if os.path.isfile(current_input_file):
+            with open(current_input_file, "r") as input_file:
+                lines = input_file.readlines()
+                file_buffer = "".join(lines)
+            with open(output_dest, "w") as output_file:
+                # Local includes must be changed to RPC framework level includes.
+                # e.g. #include "SBDefines.h" -> #include <LLDBRPC/SBDefines.h>
+                # Also, RPC common code includes must change to RPC framework level includes.
+                # e.g. #include "lldb-rpc/common/RPCPublic.h" -> #include <LLDBRPC/RPCPublic.h>
+                rpc_common_matches = RPC_COMMON_REGEX.finditer(file_buffer)
+                rpc_include_filename_matches = RPC_INCLUDE_FILENAME_REGEX.finditer(
+                    file_buffer
+                )
+                for match in rpc_common_matches:
+                    file_buffer = re.sub(
+                        match.group(),
+                        r"#include <LLDBRPC/" + match.group("include_filename") + ">",
+                        file_buffer,
+                    )
+                for match in rpc_include_filename_matches:
+                    file_buffer = re.sub(
+                        match.group(),
+                        r"#include <LLDBRPC/" + match.group("include_filename") + ">",
+                        file_buffer,
+                    )
+                output_file.write(file_buffer)
+
+
+def modify_main_includes(input_directory_path, output_directory_path):
+    for input_filepath in os.listdir(input_directory_path):
+        current_input_file = os.path.join(input_directory_path, input_filepath)
+        output_dest = os.path.join(output_directory_path, input_filepath)
+        if os.path.isfile(current_input_file):
+            with open(current_input_file, "r") as input_file:
+                lines = input_file.readlines()
+                file_buffer = "".join(lines)
+            with open(output_dest, "w") as output_file:
+                # Local includes must be changed to framework level includes.
+                # e.g. #include "lldb/API/SBDefines.h" -> #include <LLDB/SBDefines.h>
+                regex_matches = INCLUDE_FILENAME_REGEX.finditer(file_buffer)
+                for match in regex_matches:
+                    file_buffer = re.sub(
+                        match.group(),
+                        r"#include <LLDB/" + match.group("include_filename") + ">",
+                        file_buffer,
+                    )
+                output_file.write(file_buffer)
+
+
+def remove_guards(output_directory_path, unifdef_guards):
+    unifdef_path = shutil.which("unifdef")
```

Comment:
Since CMake is computing the `unifdef` binary, let's pass that in to the script. If it's not specified, we can fallback to this (e.g. for the tests).